quixstreams/app.py

```diff
@@ -431,6 +441,90 @@ def stop(self):
         """
         self._running = False
 
+    def get_producer(self) -> Producer:
+        """
+        Create and return a pre-configured Producer instance.
+        The Producer is initialized with params passed to Application.
+
+        It's useful for producing data to Kafka outside the standard Application processing flow,
+        (e.g. to produce test data into a topic).
+        Using this within the StreamingDataFrame functions is not recommended, as it creates a new Producer
+        instance each time, which is not optimized for repeated use in a streaming pipeline.
+
+        Example Snippet:
+
+        ```python
+        from quixstreams import Application
+
+        app = Application.Quix(...)
+        topic = app.topic("input")
+
+        with app.get_producer() as producer:
+            for i in range(100):
+                producer.produce(topic=topic.name, key=b"key", value=b"value")
+        ```
+        """
+        if self.is_quix_app:
+            topics = self._quix_config_builder.create_topic_configs.values()
+            if self._quix_config_builder.app_auto_create_topics:
+                self._quix_config_builder.create_topics(topics)
+            else:
+                self._quix_config_builder.confirm_topics_exist(topics)
+
+        return Producer(
+            broker_address=self._broker_address,
+            partitioner=self._partitioner,
+            extra_config=self._producer_extra_config,
+        )
+
+    def get_consumer(self) -> Consumer:
+        """
+        Create and return a pre-configured Consumer instance.
+        The Consumer is initialized with params passed to Application.
+
+        It's useful for consuming data from Kafka outside the standard Application processing flow.
+        (e.g. to consume test data from a topic).
+        Using it within the StreamingDataFrame functions is not recommended, as it creates a new Consumer instance
+        each time, which is not optimized for repeated use in a streaming pipeline.
+
+        Note: By default this consumer does not autocommit consumed offsets to allow exactly-once processing.
+        To store the offset call store_offsets() after processing a message.
+        If autocommit is necessary set `enable.auto.offset.store` to True in the consumer config when creating the app.
+
+        Example Snippet:
+
+        ```python
+        from quixstreams import Application
+
+        app = Application.Quix(...)
+        topic = app.topic("input")
+
+        with app.get_consumer() as consumer:
+            for _ in range(100):
+                msg = consumer.poll(timeout=1.0)
+                if msg is not None:
+                    # Process message
+                    # Optionally commit the offset
+                    # consumer.store_offsets(msg)
+
+        ```
+        """
+        if self.is_quix_app:
```

Comment:
not really a thing that needs changed now, but as an FYI, topic management will be a bit easier with the new TopicManager I made in a pending MR. With it, We might even consider not creating the topics unless you run an actual application, since creating topics becomes a 1-liner with it?